contentlayer.config.ts

```diff
@@ -1,18 +1,116 @@
+      name: slugifyName,
+      filePathPattern: `${name}/**/*.md`,
+      contentType: "markdown",
+      fields: {
+        title: { type: "string" },
+        speakers: { type: "list", of: { type: "string" } },
+        date: { type: "date" },
+        transcript_by: { type: "string" },
+        Transcript_by: { type: "string" },
+        "transcript by": { type: "string" },
+        categories: { type: "list", of: { type: "string" } },
+        tag: { type: "list", of: { type: "string" } },
+        tags: { type: "list", of: { type: "string" } },
+        media: { type: "string" },
+        translation_by: { type: "string" },
+        episdoe: { type: "number" },
+        episode: { type: "number" },
+        aliases: { type: "list", of: { type: "string" } },
+        vídeo: { type: "string" },
+        video: { type: "string" },
+        hosts: { type: "list", of: { type: "string" } },
+        source: { type: "string" },
+        transcription_coverage: { type: "string" },
+        summary: { type: "string" },
+        needs: { type: "string" },
+        aditional_resources: { type: "list", of: Resources },
+        additional_resources: { type: "list", of: Resources },
+        weight: { type: "number" },
+      },
+    }));
+  });
+
+  return { DefinedDocumentTypes };
+};
+
+const generateExcludedPaths = () => {
+  const otherPaths = [
+    ".github",
+    ".gitignore",
+    "LICENSE.md",
+    "README.md",
+    "twitter_handles.json",
+    ".json",
+    "2018-08-17-richard-bondi-bitcoin-cli-regtest.es.md",
+  ];
+
+  const getFolders = fs.readdirSync(path, "utf-8");
+  const folders = getFolders.slice(6).filter((item) => item !== "twitter_handles.json");
+
+  const indexFiles: string[] = [];
+  const indexFilesInFolders: string[] = [];
+
+  for (let i = 0; i < folders.length; i++) {
+    const name = folders[i];
+    const files = fs.readdirSync(`${path}/${folders[i]}`);
+
+    // isNumRegex: ** /^-?\d+$/ ** checks if file name is a digit. We're using it to check for folders that have transcripts grouped in years.
+    const isNumRegex = /^-?\d+$/;
+    const isDirectory = files.filter((num: string) => isNumRegex.test(num));
+
+    if (isDirectory.length) {
+      isDirectory.map((year) => {
+        const text = `${name}/${year}/_index.md ${name}/${year}/_index.zh.md ${name}/${year}/_index.es.md`;
+        indexFilesInFolders.push(...text.split(" "));
+      });
+    }
+  }
+
+  folders.map((folder) => {
+    const text = `${folder}/_index.md ${folder}/_index.zh.md ${folder}/_index.es.md`;
+
+    indexFiles.push(...text.split(" "));
+  });
+
+  const foldersToExclude = [...otherPaths, ...indexFiles, ...indexFilesInFolders];
+
+  return { foldersToExclude };
+};
+
+const { DefinedDocumentTypes } = generateDocuTypes();
+const { foldersToExclude } = generateExcludedPaths();
+
+export default makeSource({
+  contentDirPath: "public/bitcoin-transcript/",
+  documentTypes: DefinedDocumentTypes,
+  contentDirExclude: foldersToExclude,
+});
```

Comment:
For pulling content from a remote repository, this is the approach I am following with the glossary:
Following the approach below keeps the repository clean, as contentLayer doesn’t pull the content until build time, unlike using submodules.
However, feel free to disregard this if it doesn’t work for your case.

```suggestion
export default makeSource({
  syncFiles: syncContentFromGit,
  contentDirPath: 'public/glossary-repo',
  ...
});

const syncContentFromGit = async (contentDir: string) => {
  const syncRun = async () => {
    const gitUrl = 'https://github.com/jrakibi/glossary'
    await runBashCommand(`
      if [ -d  "${contentDir}" ];
        then
          cd "${contentDir}"; git pull;
        else
          git clone --depth 1 --single-branch ${gitUrl} ${contentDir};
      fi
    `)
  }

  let wasCancelled = false
  let syncInterval: any

  const syncLoop = async () => {
    console.log('Syncing content files from git')

    await syncRun()

    if (wasCancelled) return

    syncInterval = setTimeout(syncLoop, 1000 * 60)
  }

  // Block until the first sync is done
  await syncLoop()

  return () => {
    wasCancelled = true
    clearTimeout(syncInterval)
  }
}
```

Reply:
This is a documentation related to it:
https://contentlayer.dev/docs/sources/remote-files-fbb47906

Reply:
This is also another approach that can be followed. But I think from the last meeting we had, we agreed on pulling the data on the repository through submodules as this method might not be best for this usecase.

Reply:
@jrakibi While this approach is valid as stated in the docs and gives us similar results as with the submodule, I believe using git submodules helps us to avoid this block of code. Also, with submodules, syncing content from the parent/remote repo is a lot easier and streamlined.